gio/src/file.rs

```diff
@@ -1111,6 +1113,177 @@ pub trait FileExtManual: IsA<File> + Sized {
 
         (fut, Box::pin(receiver))
     }
+
+    #[doc(alias = "g_file_set_attribute")]
+    fn set_attribute(
+        &self,
+        attribute: &str,
+        type_value: FileAttributeTypeValue,
+        flags: FileQueryInfoFlags,
+        cancellable: Option<&impl IsA<Cancellable>>,
+    ) -> Result<(), glib::Error> {
+        unsafe {
+            let mut error = std::ptr::null_mut();
+            let is_ok = ffi::g_file_set_attribute(
+                self.as_ref().to_glib_none().0,
+                attribute.to_glib_none().0,
+                type_value.as_ref().type_().into_glib(),
+                type_value.as_ref().value().as_ptr(),
+                flags.into_glib(),
+                cancellable.map(|p| p.as_ref()).to_glib_none().0,
+                &mut error,
+            );
+            debug_assert_eq!(is_ok == glib::ffi::GFALSE, !error.is_null());
+            if error.is_null() {
+                Ok(())
+            } else {
+                Err(from_glib_full(error))
+            }
+        }
+    }
 }
 
 impl<O: IsA<File>> FileExtManual for O {}
+
+#[derive(Debug)]
+// checker-ignore-item
+pub struct FileAttributeTypeValue<'a>(FileAttributeTypeValueInner<'a>);
+
+impl<'a> FileAttributeTypeValue<'a> {
+    // new_pointer is private to gio crate to not allow creating values of arbitrary pointers from safe code
+    #[allow(dead_code)] // TODO remove this allow attribute when new_pointer will be used by this crate
+    pub(crate) fn new_pointer(type_: FileAttributeType, ptr: glib::ffi::gpointer) -> Self {
+        Self(FileAttributeTypeValueInner::Pointer(type_, ptr))
+    }
```

Comment:
i would remove this and move the dead_code to the enum if clippy complains, can be added later if needed